### ISSUE TITLE
Remove serviceSetupV2 feature flag

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#4901]
## Feature Description

Issue to remove the servicev2SetupV2 feature flag following rollout to 100% scheduled for March 14, 2022. Target date for removing the feature flag is March 28, 2022.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `serviceSetupV2` feature flag should be removed.
* All code paths that are currently run when the flag is not enabled should be removed, and all code structures (PHP classes and methods, JS components etc.) that are only used in such removed code should also be removed.
* The `Setup_V2` PHP class should be merged into its base class `Setup` so that going forward only that `Setup` class remains and is used (obviously no longer an abstract class then).
    * The PHPUnit tests should reflect that as well.
* The `Google_Proxy::SETUP_URI_V2` constant should be renamed to `Google_Proxy::SETUP_URI` (with the original `Google_Proxy::SETUP_URI` being deleted per the above).

## Implementation Brief

**Do not merge the PR for this before the 1.70.0 release has gone out, since this must wait for the 1.71.0 release.**

* Remove the `serviceSetupV2` feature flag from `feature-flags.json`
* Remove `useFeature( 'serviceSetupV2' )` from `SetupSuccessBannerNotification` and `SetupUsingProxy`. Remove all conditional checks for the flag and only render the component as if `serviceSetupV2` were enabled. Code paths where `serviceSetupV2Enabled === false` can be removed.
* Remove `createServiceSetupV2Variant` and usage from `SetupSuccessBannerNotification.stories.js`, as now all stories will default to using the V2 setup as it's the default after the above change to components. VRTs will likely need updating.
* Remove any `serviceSetupV2` feature flag setup in `SetupUsingProxy.stories.js`.
* Remove the `serviceSetupV2` feature flag usage in PHP files and remove any code paths where `Feature_Flags::enabled( 'serviceSetupV2' ) === false` in:
  * `includes/Plugin.php`
  * `includes/Core/Authentication/Authentication.php`
  * `includes/Core/Authentication/Google_Proxy.php`
  * `includes/Core/Authentication/Setup.php`
  * `includes/Core/Authentication/Clients/OAuth_Client.php`
  * `includes/Core/Modules/Module.php`
  * `includes/Modules/Analytics.php`
* Move everything from `Setup_V2` PHP class into its base class `Setup`. `Setup` no longer needs to be an abstract class.
* Rename `Google_Proxy::SETUP_URI_V2` to `Google_Proxy::SETUP_URI` (remove the original `Google_Proxy::SETUP_URI` variable).

### Test Coverage

* Any tests that rely on the `serviceSetupV2` feature being `false` should be removed/updated to use the new v2 behaviour. Tests that rely on `serviceSetupV2` being `true` can be used to replace the old tests.
* Tests around `SetupV2` and `Setup` will need updating now that `Setup` is not an abstract class.

## QA Brief

* Run through the normal site setu flows on a new site and existing site, ensuring they work without errors and are always using the new V2 Service setup.

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
@eclarke1 Added this to Sprint 71 and the 1.71.0 release (March 28) already as that needs to go in there.

**Comment by tofumatt** [comment]
Marking this as an 11 in case there are any E2E or other tests that are affected, but otherwise seems a pretty straightforward change.

**Comment by felixarntz** [comment]
@tofumatt IB almost good to go, can you please add something to the IB that clarifies or at least mentions to also address the last two points in the ACs? I guess it could be a copy of those points, or something like "Address the two last points of the ACs which are already technical detail." 😆

**Comment by tofumatt** [comment]
Oh, yes, good catch. Done!

**Comment by felixarntz** [comment]
IB ✅
